extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildBufferAggregatorHelper.java

```diff
@@ -123,7 +126,8 @@ public int getLgK()
 
   private WritableMemory getMemory(final ByteBuffer buf)
   {
-    return memCache.computeIfAbsent(buf, b -> WritableMemory.writableWrap(b, ByteOrder.LITTLE_ENDIAN));
+    return memCache.computeIfAbsent(buf,
+        b -> WritableMemory.writableWrap(b, ByteOrder.LITTLE_ENDIAN, MEM_REQ_SERVER));
```

Comment:
memory server doesn't seem to be a required  argument, do you mind explaining why we need to pass it?

Reply:
This is to restore support for quantile and HLL4 sketches to be able to grow larger than a given buffer in a buffer aggregator and move to heap in rare cases. This used to be a default before. The problem with this support arose because datasketches-memory was substantially redesigned some time ago, and the default behavior was removed. At that time we should have changed this Druid extension accordingly, but we did not. And because this support is needed in rare circumstances, the problem was not noticed for quite a while.

Reply:
Unit tests were added to make sure this support works. If you look at the changes, you can see that I changed a few tests from asserting failure to asserting success. For instance testFailureWhenMaxStreamLengthHit to testSuccessWhenMaxStreamLengthHit in DoublesSketchAggregatorTest.java

Reply:
thanks for the explanation!